packages/core/src/lib/markdown-it/index.js

```diff
@@ -75,31 +75,77 @@ markdownIt.renderer.rules.fence = (tokens, idx, options, env, slf) => {
   if (highlightLinesInput) {
     const highlightLines = highlightLinesInput.split(HIGHLIGHT_LINES_DELIMITER_REGEX);
     highlightRules = highlightLines
-      .map(ruleStr => HighlightRule.parseRule(ruleStr, -startFromZeroBased, lines))
+      .map(ruleStr => HighlightRule.parseRule(ruleStr, -startFromZeroBased, str.split('\n')))
```

Comment:
Nit/micro-optimization: maybe store the `str.split('\n')` results before the `map` so that it does not repeat the split computation every time the callback function is applied.